### ISSUE TITLE
PR #414 broke displaying packages with different name from id

### ISSUE THREAD

**Issue by adam-fowler** [swift-server/vscode-swift#425]
**Describe the bug**
SwiftPM versions using `Package.resolved` format version 1 (ie 5.5 or earlier) do not deal with packages that have the different name and id. This appears to have been caused by PR#414

**To Reproduce**
This package does not show the realm dependencies
```swift
// swift-tools-version:5.3
import PackageDescription

let package = Package(
    name: "my-package",
    products: [
        // Products define the executables and libraries a package produces, and make them visible to other packages.
        .library(name: "my-package", targets: ["my-package"]),
    ],
    dependencies: [
        .package(url: "https://github.com/realm/realm-swift", .upToNextMajor(from: "10.28.0")),
    ],
    targets: [
        .target(name: "my-package", dependencies: ["realm-swift"])
    ]
)
```
cc @stevapple

**Comment by stevapple** [comment]
I supposed #414 to only _allow more package to be displayed_, and didn’t expect this.

Will take some time to look into it.

**Comment by adam-fowler** [comment]
It is related to the Package.resolved v1 not having an identity field and it assuming it can use name instead as the identity. I was pretty sure this was working before #414 but maybe it was never working.

**Comment by adam-fowler** [comment]
See #435

**Comment by stevapple** [comment]
Given the fix I believe the real cause is #382 – hopefully we can avoid this being broken by adding a new test case.

**Comment by adam-fowler** [comment]
@stevapple Can you think of a package I can use to test this with though? Needs to be something fairly small, but heavily used. Don't really want to use Realm as it takes ages to download.

**Comment by stevapple** [comment]
@adam-fowler Do you think apple/swift-syntax is a better choice? (~10MB download)

**Comment by stevapple** [comment]
I was a bit hesitated because SwiftSyntax is under heavy development and this is subject to change. [SwiftProtobuf](https://github.com/apple/swift-protobuf) is another choice with ~25MB download.

**Comment by stevapple** [comment]
Wait, we need to support Swift 5.4 CI, right? Then SwiftSyntax and AsyncAlgorithms aren't applicable. Is SwiftProtobuf something acceptable?

**Comment by adam-fowler** [comment]
swift-cmark is fairly small. I'll use that

**Comment by adam-fowler** [comment]
Fixed with #435
